internal/magic/magic.go

```diff
@@ -107,7 +107,7 @@ func markup(sigs ...[]byte) Detector {
 		if bytes.HasPrefix(raw, []byte{0xEF, 0xBB, 0xBF}) {
 			// we restore BOM after stripping WS so its presence can be used
```

Comment:
This comment is now outdated. If you're going to replace it, please follow the [Effective Go suggestions for comments](https://go.dev/doc/effective_go#commentary), namely the one which says comments should be fully formed sentences (including capitalization and punctuation.)

Reply:
@gabriel-vasile - will do. 

I was reviewing the code and there's a discrepancy in the way that we treat the BOM in `charset.go` and in `magic.go`: in charset.go we consider all BOMs so if a document is utf16 or whatever encoded it will be detected as such - however we only skip the utf8 BOM in magic.go. What do you think of making sure we skip all the BOMs defined in `charset.go` (exporting `charset.boms` struct) to ensure consistency (although html docs in utf16 are unusual the docs allow for it - and although the docs say utf32 is not recommended it isn't prohibited either)?

Reply:
I'm not sure, please give me a day or two to think about it. 
The thing is, the HTML detection in `magic.go` follows the rules in this table: [MIME sniffing specification](https://mimesniff.spec.whatwg.org/#identifying-a-resource-with-an-unknown-mime-type) and **it only works for UTF8 documents**. An UTF16 HTML document will fail to be detected as HTML and fall back to `text/plain`. This means, trimming all BOMs is a step forward but the results are still the same.

On the other hand, there is a point in not having the BOMs defined in each package...

Reply:
@napalu I think it's ok they way it is now, i.e. trimming only UTF8 BOM.

So the only thing left is updating [this comment](https://github.com/gabriel-vasile/mimetype/blob/e4f5c6c96533c3e35b8bd01f5161308cba48576c/internal/magic/magic.go#L107-L109) to match the code. Please mention in your comment that the 3 bytes slice is the UTF8 BOM.

Reply:
@gabriel-vasile done.